scripts/vscripts/tf2ware_ultimate/minigames/pick_up_that_can.nut

```diff
@@ -0,0 +1,222 @@
+minigame <- Ware_MinigameData
+({
+	name          = "Pick Up That Can"
+	author		  = "CzechMate"
+	description   = "Pick Up And Trash The Can!"
+	duration      = 7.0
+	music         = "sweetdays"
+})
+
+::trashcan_model <- "models/props_trainstation/trashcan_indoor001b.mdl"
+::can_model <- "models/props_junk/popcan01a.mdl"
+
+////////////////////////// --- Settings --- ///////////////////////////////////////////
+
+const PickupDistance = 100.0; // Distance the player can pick up the prop from
+const Distance = 60.0;		// Distance from player to the picked up prop
+const Outline = true;       // Outline props?
+const CanOffset = 100.0;       // X and Y Offset for spawning cans
+const CanHeightOffset = 300.0; // Z Offset for spawning cans
+const PropFaceSameDirection = false // If true, makes the prop face the same direction as the player (similar to Half-Life 2)
+local pickableObjects = ["prop_physics"]; // List of classnames that can be picked up
+
+//////////////////////////////////////////////////////////////////////////////////////
+
+function OnPrecache()
+{
+	PrecacheModel(can_model)
+	PrecacheModel(trashcan_model)
+}
+
+function OnStart()
+{
+	foreach (player in Ware_MinigamePlayers)
+	{
+            player.ValidateScriptScope();
+            player.GetScriptScope().LastButtons <- 0;

```

Comment:
Do not use GetScriptScope, instead use Ware_GetPlayerMiniData to store variables (it deletes all variables in that table when minigame ends automatically). Keep it for the PlayerThink below though as it won't work otherwise